src/RealizationOrder.cpp

```diff
@@ -40,11 +40,16 @@ vector<vector<string>> find_fused_groups(const vector<string> &order,
     set<string> visited;
     vector<vector<string>> result;
 
-    for (const auto &fn : order) {
+    // Iterate starting from the last function to be realized. This way,
+    // if any of the functions in the fused group refers to other functions
+    // that were scheduled inline, those inlined function will come first
+    // (first to be realized) before the fused group in the fuse_adjacency_list.
+    for (int i = (int)order.size() - 1; i >= 0; --i) {
+        const string &fn = order[i];
         if (visited.find(fn) == visited.end()) {
             vector<string> group;
             find_fused_groups_dfs(fn, fuse_adjacency_list, visited, group);
-            result.push_back(group);
+            result.insert(result.begin(), group);
```

Comment:
This is potentially a really inefficient thing to do (inserting at the beginning of a vector requires moving everything in the vector). It *probably* doesn't matter because the set of functions probably can't ever be big enough to really be a problem...

But, it has an easy fix: simply push_back in the loop, and then use std::reverse before returning.